inspirehep/celery.py

```diff
@@ -25,15 +25,27 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
+import os
 
-from flask_celeryext import create_celery_app
+from flask_celeryext import AppContextTask, create_celery_app
+from sqlalchemy.exc import InvalidRequestError, OperationalError
 
-from .factory import create_app
+from inspirehep.factory import create_app
 
 
-celery = create_celery_app(
-    create_app(LOGGING_SENTRY_CELERY=True)
-)
+LOGGER = logging.getLogger(__name__)
+
+
+class CeleryTask(AppContextTask):
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        if isinstance(exc, InvalidRequestError) or isinstance(exc, OperationalError):
+            LOGGER.error('Shutting down celery process because of {}', format(exc))
```

Comment:
* are you sure you can do formatting like this?
```ipython
In [10]: logger.error('foo {}', format('bar'))                                                                                                                                                                       
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 994, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 840, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 577, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/micha/.local/bin/ipython", line 10, in <module>
    sys.exit(start_ipython())
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/__init__.py", line 125, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/terminal/ipapp.py", line 356, in start
    self.shell.mainloop()
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/terminal/interactiveshell.py", line 498, in mainloop
    self.interact()
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/terminal/interactiveshell.py", line 489, in interact
    self.run_cell(code, store_history=True)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 2855, in run_cell
    raw_cell, store_history, silent, shell_futures)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 2881, in _run_cell
    return runner(coro)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/async_helpers.py", line 68, in _pseudo_sync_runner
    coro.send(None)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 3058, in run_cell_async
    interactivity=interactivity, compiler=compiler, result=result)
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 3249, in run_ast_nodes
    if (await self.run_code(code, result,  async_=asy)):
  File "/home/micha/.local/pipx/venvs/ipython/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 3326, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-10-a0864fec6e7d>", line 1, in <module>
    logger.error('foo {}', format('bar'))
Message: 'foo {}'
Arguments: ('bar',)
```
* no structured logging?
* if you do `LOGGER.exception` instead the stack trace goes to sentry.

Reply:
This is `next`, we don't have structured logs...

Reply:
right, my bad. I guess we'll want this on `hep` too though.

Reply:
Yep

Reply:
regarding the `format`, it was a typo.